### PR TITLE
Add tests for Number#toPrecision

### PR DESCRIPTION
**Pull request by leobalter**
I'm not happy with the lack of separation for the tests on return-values.js, but it sounded less confusing this way.

### CODE REVIEW

File: test/built-ins/Number/prototype/toPrecision/nan.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+assert.sameValue(n.toPrecision(p), "NaN", "Number NaN");
+assert.sameValue(calls, 1, "Number NaN is checked after ToInteger(precision)");
+
+calls = 0;
+assert.sameValue(NaN.toPrecision(undefined), "NaN");
+assert.sameValue(
+  calls,
+  0,
+  "returned before ToInteger(precision) if precision is undefined"
+);
```
Comment (leobalter) [inline test/built-ins/Number/prototype/toPrecision/nan.js]: this is wrong, I need to fix it.

**Comment by leobalter** [comment]
pushed a few fixes/updates

**Comment by jugglinmike** [comment]
There's some duplication here, making the tests a little noisy (e.g. multiple
tests for step 1, a few assertions about the [[NumberData]] of
Number.prototype), but nothing worth blocking on. I'll merge this shortly!

**Comment by jugglinmike** [comment]
Squashed and merged to `master` at commit 60e61a15e0706eae864fe29991df3a6e8c8a3c28. Thanks, Leo!

**Pull request closed without merging**